nss/integration-tests/helper_test.go

```diff
@@ -122,17 +122,19 @@ paths:
 		cmd.Env = append(cmd.Env, fmt.Sprintf("GOCOVERDIR=%s", rawCovDir))
 	}
 
+	stopped = make(chan struct{})
 	go func() {
 		require.NoError(t, cmd.Start(), "Setup: daemon should start with no error")
 		<-ctx.Done()
-		require.NoError(t, cmd.Process.Kill(), "Teardown: daemon should exit with no error")
-		_ = cmd.Wait()
+		require.NoError(t, cmd.Process.Signal(os.Signal(syscall.SIGTERM)), "Teardown: Failed to send signal to stop daemon")
```

Comment:
I know what this is about because I have the context from the other comment, but I think a comment on why we are sending SIGTERM and wait for it would be beneficial :)

Another way: if we run with CommandContext, does cancelling the context sends SIGTERM? (and then, we just have to wait)

Reply:
I just checked: `CommandContext` sends SIGKILL when the context is canceled, so it does not work for us.

I'll add a comment explaining why we send SIGTERM specifically.